### PR TITLE
Add extended quadword BCD <> binary conversion to vec_bcd_ppc.h:

### PR DESCRIPTION
**Pull request by munroesj52**
Documnent extened BCD <-> binary conversions with examples.
Add a POWER9 specific optimization to vec_bcdctuq().

	* src/vec_bcd_ppc.h: Improve Doxygen overview.
	Cross reference BCD conversion and quadword long division
	from vec_int128_ppc.h
	[subsubsection bcd128_convert_0_2_3]: Add section
	"Multiple precision BCD to/from Binary conversion"
	[\paragraph bcd128_convert_0_2_3_0]: Add section
	"Multiple precision BCD from Binary conversion"
	[\paragraph bcd128_convert_0_2_3_1]: Add section
	"Multiple precision BCD to Binary conversion"
	(vec_bcdctuq [_ARCH_PWR9]): Optimization.

Signed-off-by: Steven Munroe <munroesj52@gmail.com>

### CODE REVIEW

File: src/vec_bcd_ppc.h [hunk 1]
```diff
@@ -1109,15 +1110,54 @@ example_vec_bcdctuq (vui8_t vra)
...
+  vrt = (vui128_t) vec_bcdctsq (sbcd);
+  // Then X 10 plus the unit digit to complete 32-digit convert
+  vrt = vec_mul10euq (vrt, vrd);
+#else
+  // P7/P8 implementation as above
+#endif
+  return vrt;
+}
+ * \endcode
+ * This add a few more cycles to split the high digits from the low
```
Comment (ThinkOpenly) [inline src/vec_bcd_ppc.h]: s/add/adds/

File: src/vec_bcd_ppc.h [hunk 2]
```diff
@@ -881,18 +881,18 @@ vec_bcddiv (vBCD_t a, vBCD_t b)
...
+ * better than the serial conversion examples described above.
  *
  * But part of the value of PVECLIB is to provide support across
  * POWER7/8/9 and across compiler versions.
  * The convert instructions above are not supported in current
- * compilers with built-ins so PVECLIB should provide in-line
+ * compilers with built-ins so PVECLIB provides in-line
  * assembler implementations for these operations.
- * Now we should looking into better algorithms for implementing
+ * Now we need look into better algorithms to implementing
```
Comment (ThinkOpenly) [inline src/vec_bcd_ppc.h]: I would've kept "for" here, instead of "to", but it's subjective.

File: src/vec_bcd_ppc.h [hunk 3]
```diff
@@ -1332,6 +1372,170 @@ vec_bcdcfuq (vui128_t vra)
...
  * \endcode
+ * \note This process can be extended to 256, 512, 1024-bits, etc by
+ * widening the first 5 steps appropriately and adding steps using
+ * extended quadword multiply and add operations from vec_int128_ppc.h
+ * (\ref int128_examples_0_1_3_1).
+ *
+ * \subsubsection bcd128_convert_0_2_3 Multiple precision BCD to/from Binary conversion
+ *
+ * The simplest case is converting a vector unsigned __int128 to BCD.
+ * his requires up to 39 digits across two vectors. This can either
```
Comment (ThinkOpenly) [inline src/vec_bcd_ppc.h]: s/his/This/
s/either/either be/

File: src/vec_bcd_ppc.h [hunk 4]
```diff
@@ -1332,6 +1372,170 @@ vec_bcdcfuq (vui128_t vra)
...
+ * passed as the dividend to the next step, produce additional 32-digit
+ * groups. This continues until the quotient is less then the divisor
+ * (in this case 10<SUP>32</SUP>). This final quadword quotient
+ * provides the highest order 32-digit group for the conversion.
+ * The digit groups are produced in order from lowest to highest
+ * significance.
+ *
+ * As the conversion process continues the number of quadwords in the
+ * extended dividend/quotient shrinks. The divide / modulo quadword
+ * by constant operations, test for leading zeros and skip over them.
```
Comment (ThinkOpenly) [inline src/vec_bcd_ppc.h]: delete the comma.

File: src/vec_bcd_ppc.h [hunk 5]
```diff
@@ -1332,6 +1372,170 @@ vec_bcdcfuq (vui128_t vra)
...
+    }
+
+  return cnt;
+}
+ * \endcode
+ * This process starts with a single quadword (the converted high order
+ * digit group). As additional digit groups are converted, the extended
+ * binary value is multiplied by 10<SUP>32</SUP> before adding the
+ * converted digit group.  The number of quadwords in the array
+ * <I>d[]</I> expand as need to hold the binary value.
```
Comment (ThinkOpenly) [inline src/vec_bcd_ppc.h]: s/as need/as needed/

File: src/vec_bcd_ppc.h (hunk 1 above)
Comment (munroesj52) [reply src/vec_bcd_ppc.h]: yes

File: src/vec_bcd_ppc.h (hunk 2 above)
Comment (munroesj52) [reply src/vec_bcd_ppc.h]: yes

File: src/vec_bcd_ppc.h (hunk 3 above)
Comment (munroesj52) [reply src/vec_bcd_ppc.h]: yes

File: src/vec_bcd_ppc.h (hunk 4 above)
Comment (munroesj52) [reply src/vec_bcd_ppc.h]: yes

File: src/vec_bcd_ppc.h (hunk 5 above)
Comment (munroesj52) [reply src/vec_bcd_ppc.h]: yes

**Pull request merged**